### ISSUE TITLE
Terraform code is failing and throwing error when I query available provider package

### ISSUE THREAD

**Issue by NeevArya** [Azure/terraform-provider-azapi#210]
I am trying to use the Azure/Azapi Provider within my Terraform project to create Key Vault's key rotation policy but after I add the provider and run terraform init, I get the following error alternatively:

#1 │ Error: Failed to install provider │ │ Error while installing azure/azapi v0.1.0: could not query provider registry for registry.terraform.io/azure/azapi: failed to
│ retrieve authentication checksums for provider: 403 Forbidden

#2 │ Error: Failed to install provider │ │ Error while installing azure/azapi v0.1.0: could not query provider registry for registry.terraform.io/azure/azapi: failed to
│ retrieve authentication checksums for provider: the request failed after 2 attempts, please try again later: Get │ "https://github.com/Azure/terraform-provider-azapi/releases/download/v0.1.0/terraform-provider-azapi_0.1.0_SHA256SUMS": net/http:
│ request canceled (Client.Timeout exceeded while awaiting headers)

_providers.tf_
provider "azurerm" {
  features {}
}
provider "azuread" {
}
provider "azapi" {
}

_versions.tf_

`terraform {
  required_providers {
    azurerm = {
      source = "hashicorp/azurerm"
      version = ">= 2.7.0"
    }
    azuread = {
      source  = "hashicorp/azuread"
      version = "2.29.0"
    }
    azapi = {
      source = "azure/azapi"
      version = "=0.1.0"
    }
  }
}`

Do you guys have any idea why its failing to install the azapi provider and rest azurerm, azuread are working perfectly fine.

**Comment by mlcooper** [comment]
I noticed you're trying to install version 0.1.0 which is extremely old. I recommend trying 1.0.0.

**Comment by NeevArya** [comment]
@mlcooper Actually I have tried installing all the other versions too, from 0.1.0 to 1.0.0 , every version is returning the same error.

**Comment by ms-henglu** [comment]
Hi @NeevArya,

Thank you for taking time to report this issue!

Do you have some firewall rules that restricted you from accessing GitHub? Because this provider is downloaded from GitHub.

**Comment by arkiaconsulting** [comment]
same here, and no firewall rules are set.
Terraform version: 1.1.7
azapi version 1.0.0

I tried to delete the `.terraform` folder, the `.terraform.lock.hcl`

It fails with:
```
Error: Failed to query available provider packages
│
│ Could not retrieve the list of available versions for provider hashicorp/azapi: provider registry registry.terraform.io does not have a provider named registry.terraform.io/hashicorp/azapi
│
│ Did you intend to use azure/azapi? If so, you must specify that source address in each module which requires that provider. To see which modules are currently depending on hashicorp/azapi, run
│ the following command:
│     terraform providers

**Comment by arkiaconsulting** [comment]
> 

forget about it. Was a typo.... 🤒

**Comment by NeevArya** [comment]
@arkiaconsulting can you please elaborate about the typo you mentioned?

**Comment by arkiaconsulting** [comment]
shame on me ^^ (azpi <=> azapi)

```hcl
terraform {
  required_providers {    
    azpi = {      
      source  = "azure/azapi"
      version = "=1.0.0"
    }
  }
}

provider "azapi" {
  default_location = local.location
  subscription_id = local.specific.apim.subscription_id
}
```

**Comment by ms-henglu** [comment]
Hi @NeevArya , is this issue resolved with azapi v1.0.0?

**Comment by ms-henglu** [comment]
I'll close this issue since there's no response, but feel free to reopen it.
